plugin.video.themoviedb.helper/script.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# Module: default
+# Author: jurialmunkey
+# License: GPL v.3 https://www.gnu.org/copyleft/gpl.html
+import sys
+import xbmc
+import xbmcgui
+import xbmcaddon
+import time
+import resources.lib.utils as utils
+from resources.lib.globals import LANGUAGES, APPEND_TO_RESPONSE
+from resources.lib.tmdb import TMDb
+_homewindow = xbmcgui.Window(10000)
+_prefixname = 'TMDbHelper.'
+_addon = xbmcaddon.Addon()
+_addonname = 'plugin.video.themoviedb.helper'
+_languagesetting = _addon.getSetting('language')
+_language = LANGUAGES[int(_languagesetting)]
+_cache_long = int(_addon.getSetting('cache_details_days'))
+_cache_short = int(_addon.getSetting('cache_list_days'))
+_tmdb_apikey = _addon.getSetting('tmdb_apikey')
+_tmdb = TMDb(api_key=_tmdb_apikey, language=_language, cache_long=_cache_long, cache_short=_cache_short,
+             append_to_response=APPEND_TO_RESPONSE, addon_name=_addonname)
+
+
+class Script:
+    def __init__(self):
+        self.params = {}
+        self.prefixpath = '{0}Path.'.format(_prefixname)
+        self.prefixlock = '{0}Locked'.format(_prefixname)
+        self.prefixcurrent = '{0}Current'.format(self.prefixpath)
+        self.prefixposition = '{0}Position'.format(_prefixname)
+        self.position = _homewindow.getProperty(self.prefixposition)
+        self.position = int(self.position) if self.position else 0
+        self.prevent_del = _homewindow.getProperty(self.prefixlock)
+        self.prevent_del = True if self.prevent_del else False
+
+    def get_params(self):
+        for arg in sys.argv:
+            if arg == 'script.py':
+                pass
+            elif '=' in arg:
+                arg_split = arg.split('=', 1)
+                if arg_split[0] and arg_split[1]:
+                    key, value = arg_split
+                    self.params.setdefault(key, value)
+            else:
+                self.params.setdefault(arg, True)
+
+    def reset_props(self):
+        _homewindow.clearProperty(self.prefixcurrent)
+        _homewindow.clearProperty(self.prefixposition)
+        _homewindow.clearProperty('{0}0'.format(self.prefixpath))
+        _homewindow.clearProperty('{0}1'.format(self.prefixpath))
+
+    def set_props(self, position=1, path=''):
+        _homewindow.setProperty(self.prefixcurrent, path)
+        _homewindow.setProperty('{0}{1}'.format(self.prefixpath, position), path)
+        _homewindow.setProperty(self.prefixposition, str(position))
+
+    def lock_path(self, condition):
+        if condition:
+            _homewindow.setProperty(self.prefixlock, 'True')
+        else:
+            self.unlock_path()
+
+    def unlock_path(self):
+        _homewindow.clearProperty(self.prefixlock)
+
+    def call_window(self):
+        sleeper = float(self.params.get('delay', '0'))
+        time.sleep(sleeper)
+        if self.params.get('call_id'):
+            xbmc.executebuiltin('Dialog.Close(all, force)')
```

Comment:
This is a brute force command it will close all dialogs even those unrelated to this script. Please specify a window id to close.